pkgs/top-level/python-packages.nix

```diff
@@ -6952,19 +6952,22 @@ in modules // {
     };
   };
 
-  lti = buildPythonPackage rec {
-    version = "0.4.0";
+  lti = let self' = (self.override {self = self';}) // {pytest = self.pytest_27;};
+  in buildPythonPackage rec {
+    version = "0.4.1";
     name = "PyLTI-${version}";
 
+    disabled = !isPy27;
+
     propagatedBuildInputs = with self; [ httplib2 oauth oauth2 semantic-version ];
-    buildInputs = with self; [
+    buildInputs = with self'; [
       flask httpretty oauthlib pyflakes pytest pytestcache pytestcov covCore
       pytestflakes pytestpep8 sphinx mock
     ];
```

Comment:
Reply to https://github.com/NixOS/nixpkgs/pull/16086/files#r67858876:
It does not work because the buildInputs also contain `pytestcache pytestcov pytestflakes pytestpep8`. As these depend on pytest_29, a conflict occurs if the override is not deep enough.
That being said, I override pytest only for buildInputs, not propagatedBuildInputs, and while such an override looks big, it is really a very small overhead thanks to lazyness.